Sources/SwiftSyntaxMacrosTestSupport/Assertions.swift

```diff
@@ -0,0 +1,305 @@
+  XCTAssertEqual(location.column, spec.column, "column does not match", file: spec.originatorFile, line: spec.originatorLine)
+
+  XCTAssertEqual(spec.severity, diag.diagMessage.severity, "severity does not match", file: spec.originatorFile, line: spec.originatorLine)
+
+  if let highlight = spec.highlight {
+    var highlightedCode = ""
+    highlightedCode.append(diag.highlights.first?.with(\.leadingTrivia, []).description ?? "")
+    for highlight in diag.highlights.dropFirst().dropLast() {
+      highlightedCode.append(highlight.description)
+    }
+    if diag.highlights.count > 1 {
+      highlightedCode.append(diag.highlights.last?.with(\.trailingTrivia, []).description ?? "")
+    }
+
+    assertStringsEqualWithDiff(
+      highlightedCode.trimmingTrailingWhitespace(),
+      highlight.trimmingTrailingWhitespace(),
+      "highlight does not match",
+      file: spec.originatorFile,
+      line: spec.originatorLine
+    )
+  }
+  if diag.notes.count != spec.notes.count {
+    XCTFail(
+      """
+      Expected \(spec.notes.count) notes but received \(diag.notes.count):
+      \(diag.notes.map(\.debugDescription).joined(separator: "\n"))
+      """,
+      file: spec.originatorFile,
+      line: spec.originatorLine
+    )
+  } else {
+    for (note, expectedNote) in zip(diag.notes, spec.notes) {
+      assertNote(note, in: tree, expected: expectedNote)
+    }
+  }
+  if diag.fixIts.count != spec.fixIts.count {
+    XCTFail(
+      """
+      Expected \(spec.fixIts.count) Fix-Its but received \(diag.fixIts.count):
+      \(diag.fixIts.map(\.message.message).joined(separator: "\n"))
+      """,
+      file: spec.originatorFile,
+      line: spec.originatorLine
+    )
+  } else {
+    for (fixIt, expectedFixIt) in zip(diag.fixIts, spec.fixIts) {
+      assertFixIt(fixIt, expected: expectedFixIt)
+    }
+  }
+}
+
+/// Assert that expanding the given macros in the original source produces
+/// the given expanded source code.
+///
+/// - Parameters:
+///   - originalSource: The original source code, which is expected to contain
+///     macros in various places (e.g., `#stringify(x + y)`).
+///   - expandedSource: The source code that we expect to see after performing
+///     macro expansion on the original source.
+///   - diagnostics:
+///   - macros: The macros that should be expanded, provided as a dictionary
+///     mapping macro names (e.g., `"stringify"`) to implementation types
+///     (e.g., `StringifyMacro.self`).
+///   - testModuleName: The name of the test module to use.
+///   - testFileName: The name of the test file name to use.
+public func assertMacroExpansion(
+  _ originalSource: String,
+  expandedSource: String,
+  diagnostics: [DiagnosticSpec] = [],
+  macros: [String: Macro.Type],
+  testModuleName: String = "TestModule",
+  testFileName: String = "test.swift",
+  indentationWidth: Trivia = .spaces(4),
```

Comment:
What are your thoughts on `indentWidth` vs `indentationWidth`?

Reply:
I don’t have strong feelings. Do you?

Reply:
I just checked. SwiftSyntax general prefers `indentation` over `indent`, so I’d prefer to stick with `indentation` for consistency.